airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/python_connectors.py

```diff
@@ -197,48 +163,32 @@ class IntegrationTests(PytestStep):
     test_directory_name = "integration_tests"
     bind_to_docker_host = True
 
-
-async def run_all_tests(context: ConnectorContext) -> List[StepResult]:
-    """Run all tests for a Python connector.
-
-    Args:
-        context (ConnectorContext): The current connector context.
-
-    Returns:
-        List[StepResult]: The results of all the steps that ran or were skipped.
+def get_test_steps(context: ConnectorContext) -> List[StepToRun]:
     """
-    step_results = []
-    build_connector_image_results = await BuildConnectorImages(context, LOCAL_BUILD_PLATFORM).run()
-    if build_connector_image_results.status is StepStatus.FAILURE:
-        return [build_connector_image_results]
-    step_results.append(build_connector_image_results)
-
-    connector_container = build_connector_image_results.output_artifact[LOCAL_BUILD_PLATFORM]
-
-    context.connector_secrets = await secrets.get_connector_secrets(context)
-
-    unit_test_results = await UnitTests(context).run(connector_container)
-
-    if unit_test_results.status is StepStatus.FAILURE:
-        return step_results + [unit_test_results]
-    step_results.append(unit_test_results)
-    async with asyncer.create_task_group() as task_group:
-        tasks = [
-            task_group.soonify(IntegrationTests(context).run)(connector_container),
-            task_group.soonify(AcceptanceTests(context, context.concurrent_cat).run)(connector_container),
-            task_group.soonify(CheckBaseImageIsUsed(context).run)(),
-        ]
-
-    return step_results + [task.value for task in tasks]
-
-
-async def run_code_format_checks(context: ConnectorContext) -> List[StepResult]:
-    """Run the code format check steps for Python connectors.
-
-    Args:
-        context (ConnectorContext): The current connector context.
-
-    Returns:
-        List[StepResult]: Results of the code format checks.
+    Get all the tests steps for a Python connector.
     """
-    return [await CodeFormatChecks(context).run()]
+    return [
+        [StepToRun(id="build", step=BuildConnectorImages(context, LOCAL_BUILD_PLATFORM))],
+        [StepToRun(
+            id="unit",
+            step=UnitTests(context),
+            args=lambda results: {"connector_under_test": results["build"].output_artifact[LOCAL_BUILD_PLATFORM]},
+            depends_on=["build"],
```

Comment:
If I'm not mistaken the dependency is always implicitly defined by the fact that a Step args are from a dependency `output_artifact`. Could we infer the dependency tree dynamically from the args then?
I know explicit is better than implicit 😄 , so feel free to reject this suggestion

Reply:
Thats fair! but there sometimes output that may not be an artifact, thus theres no implicit relationship either.

e.g. we have a step called `PushToDockerHub` and a step called `VerifyOnDockerHub`